### PR TITLE
MINOR: Fix missing UNREGISTER_BROKER call in KafkaApis

### PR DESCRIPTION
**Pull request by cmccabe**
Fix some bugs in the KRaft unregisterBroker API and add a junit test.
    
1. kafka-cluster-tool.sh unregister should fail if no broker ID is passed.
    
2. UnregisterBrokerRequest must be marked as a KRaft broker API so that KRaft brokers can receive it.
    
3. KafkaApis.scala must forward UNREGISTER_BROKER to the controller.

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -229,6 +229,8 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.ENVELOPE => handleEnvelope(request, requestLocal)
         case ApiKeys.DESCRIBE_CLUSTER => handleDescribeCluster(request)
         case ApiKeys.DESCRIBE_PRODUCERS => handleDescribeProducersRequest(request)
+        case ApiKeys.UNREGISTER_BROKER => maybeForwardToController(request,
```
Comment (dengziming) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: nit: we already have a `forwardToControllerOrFail` method which can be used here.

Comment (dajac) [reply core/src/main/scala/kafka/server/KafkaApis.scala]: +1

File: core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala [hunk 2]
```diff
@@ -725,4 +726,56 @@ class KRaftClusterTest {
...
+      cluster.format()
+      cluster.startup()
+      cluster.waitForReadyBrokers()
+      TestUtils.waitUntilTrue(() => brokerIsUnfenced(clusterImage(cluster, 1), 0),
+        "Timed out waiting for broker 0 to be unfenced.")
+      cluster.brokers().get(0).shutdown()
+      TestUtils.waitUntilTrue(() => !brokerIsUnfenced(clusterImage(cluster, 1), 0),
+        "Timed out waiting for broker 0 to be fenced.")
+      val admin = Admin.create(cluster.clientProperties())
+      try {
```
Comment (dajac) [inline core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala:770]: nit: We could use `TestUtils.resource` for auto-closing the admin and the cluster. Not a big win so I leave it up to you.

**Review by dajac — COMMENTED** [review]
Thanks for the PR. Left two minor comments.

File: core/src/main/scala/kafka/server/KafkaApis.scala (hunk 1 above)
Comment (cmccabe) [reply core/src/main/scala/kafka/server/KafkaApis.scala]: good idea

**Review by hachikuji — APPROVED** [review]
Thanks, LGTM

File: core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala (hunk 2 above)
Comment (cmccabe) [reply core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala:770]: Hmm, the try..catch block seems more like normal Scala style to me. So I'll leave it as-is for now...

**Pull request merged**